tests/std/tests/GH_002058_debug_iterator_race/test.cpp

```diff
@@ -0,0 +1,51 @@
+#include <future>
+#include <list>
+#include <set>
+#include <vector>
+
+using namespace std;
+
+// Concurrently destroy iterators and invalidate iterators
+template <class container>
+void test_concurrent_destruction() {
+    container c;
+    c.insert(c.begin(), 0);
+
+    vector<typename container::iterator> iters(1000, c.begin());
+    {
+        auto destroyIters = async(launch::async, [&]() { iters.clear(); });
+
+        auto invalidateIters = async(launch::async, [&]() { c.clear(); });
+    }
+}
+
+// Concurrently create iterators and invalidate iterators

```

Comment:
Repeating "iterators" sounds weird.
```suggestion
// Concurrently create and invalidate iterators
```